lightpath/ui/widgets.py

```diff
@@ -177,21 +180,52 @@ def add_signal(self, signal):
         self.command_layout.insertWidget(count - 1, label, 0, Qt.AlignHCenter)
         self.command_layout.insertWidget(count, widget, 0, Qt.AlignHCenter)
 
+    def get_state_color(self) -> QColor:
+        """
+        Determine the icon color given the device state and path status
+        If device is an impediment: state_color['blocking']
+        If device is removed: state_color['removed']
+        If device is in, not blocking (mirrors): state_color['half_removed']
+        If device is unknown / errored: state_color['error']
+
+        The color of the labels should quickly point users to blocking devices,
+        while providing useful information about each device's state.
+
+        Could take a color map in the future?  Colorblind support?
+
+        Returns
+        -------
+        QColor
+            the color to apply to the icon
+        """
+        device_state, _ = find_device_state(self.device)
```

Comment:
I'm not 100% sure about this, but I think this bit might make the lightpath first load go painfully slow when devices are disconnected?

Something is pathing through https://github.com/pcdshub/lightpath/blob/05a9394e09acf256649738f6233d54a7ba2b1b4d/lightpath/path.py#L83 at startup for all devices

Might be good to look at this in a follow-up

Reply:
What I'm trying to say is: all disconnected devices are timing out at one of these function calls in series before any device is shown in lightpath

Reply:
It might worse than this actually, I can't seem to switch to a path that has a disconnected device
I just see this forever and my ui freezes
```
[2022-08-24 16:36:19] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:21] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:29] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:31] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:39] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:41] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:49] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:51] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:36:59] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:01] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:09] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:11] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:19] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:21] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:29] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:31] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:39] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:41] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:49] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:51] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:37:59] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:38:01] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:38:09] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:38:11] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
[2022-08-24 16:38:19] - WARNING -  Unable to connect to BeckhoffSlits(SL2K2:SCATTER, name=sl2k2)
```

Reply:
This makes sense.  I think the real root of the issue is the fact that I increased the default EpicsSignal timeout to 10s, otherwise we'd get a slew of timeout errors at startup.  

Each `LightRow` widget updates its state once after they are created, which calls `find_device_state`.  

Something I considered doing was simply making timeout errors less severe, and assuming that devices would connect eventually.  That seemed dangerous though.

Reply:
> Each LightRow widget updates its state once after they are created, which calls find_device_state.

Somehow there must be at least one other place this is done that blocks qt from doing anything forever